common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -606,28 +621,25 @@ def get_demand_hint(self, hint_index):
         event_info['hint_text'] = get_inner_html_from_xpath(demand_hints[hint_index])
         self.runtime.publish(self, 'edx.problem.hint.demandhint_displayed', event_info)
 
+        _, should_enable_next_hint = self._should_enable_demand_hint(hint_index, demand_hints)
+
         # We report the index of this hint, the client works out what index to use to get the next hint
         return {
             'success': True,
             'hint_index': hint_index,
-            'html': self.get_problem_html(encapsulate=False, demand_hint_text=total_text, hint_index=hint_index)
+            'should_enable_next_hint': should_enable_next_hint,
+            'msg': total_text,
         }
 
-    def get_problem_html(self, encapsulate=True,
-                         demand_hint_text=None,
-                         hint_index=0,
-                         save_notification_message=None,
-                         submit_notification=False):
+    def get_problem_html(self, encapsulate=True, submit_notification=False, hint_index=0):
```

Comment:
We no longer need the hint_index variable (default value of 0 will need to be assumed in the code).